src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -10863,8 +10870,10 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
             // targeting the return value.
             let tgtEx = if isRet then enum 0 else AttributeTargets.ReturnValue
             let attrs, _ = TcAttributesMaybeFailEx false cenv envinner tgt tgtEx attrs
+            let attrs: Attrib list = attrs
             if attrTgt = enum 0 && not (isNil attrs) then
-                errorR(Error(FSComp.SR.tcAttributesAreNotPermittedOnLetBindings(), mBinding))
+                for attr in attrs do
```

Comment:
Instead of showing the overall range. We will now only show the specific attribute name